tests/csapi/sync_timeline_test.go

```diff
@@ -0,0 +1,365 @@
+package csapi_tests
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"reflect"
+	"testing"
+
+	"github.com/tidwall/gjson"
+
+	"github.com/matrix-org/complement/internal/b"
+	"github.com/matrix-org/complement/internal/client"
+	"github.com/matrix-org/complement/internal/match"
+	"github.com/matrix-org/complement/internal/must"
+)
+
+func TestSyncTimeline(t *testing.T) {
+	deployment := Deploy(t, b.BlueprintAlice)
+	defer deployment.Destroy(t)
+	alice := deployment.Client(t, "hs1", "@alice:hs1")
+
+	reqBody, err := json.Marshal(map[string]interface{}{
+		"room": map[string]interface{}{
+			"timeline": map[string]interface{}{
+				"limit": 1,
+			},
+		},
+	})
+	if err != nil {
+		t.Fatalf("unable to marshal filter: %v", err)
+	}
+
+	filterLimit1ID := createFilter(t, alice, reqBody, alice.UserID)
+
+	t.Run("Parallel", func(t *testing.T) {
+		// sytest: Can sync a room with a single message
+		t.Run("Can sync a room with a single message", func(t *testing.T) {
+			t.Parallel()
+			reqBody, err = json.Marshal(map[string]interface{}{
+				"room": map[string]interface{}{
+					"timeline": map[string]int{
+						"limit": 2,
+					},
+				},
+			})
+			if err != nil {
+				t.Fatalf("unable to marshal filter: %v", err)
+			}
+
+			filterID := createFilter(t, alice, reqBody, alice.UserID)
+			roomID := alice.CreateRoom(t, struct{}{})
```

Comment:
Never do this as Synapse and Dendrite have different defaults when there is no preset. Always say if you want this to be public/private.